src/pages/home/report/ReportActionItem.js

```diff
@@ -117,6 +120,21 @@ function ReportActionItem(props) {
         focusTextInputAfterAnimation(textInputRef.current, 100);
     }, [isDraftEmpty]);
 
+    // Hide the message if it is being moderated for a higher offense, or is hidden by a moderator
+    // Removed messages should not be shown anyway and should not need this flow
+    useEffect(() => {
+        if (!props.action.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT || _.isEmpty(props.action.message[0].moderationDecisions)) {
+            return;
+        }
+
+        // Right now we are only sending the latest moderationDecision to the frontend even though it is an array
+        const latestDecision = props.action.message[0].moderationDecisions[0];
+        if (latestDecision.decision === CONST.MODERATION.MODERATOR_DECISION_PENDING_HIDE || latestDecision.decision === CONST.MODERATION.MODERATOR_DECISION_HIDDEN) {
+            setIsHidden(true);
+        }
+        setModerationDecision(latestDecision.decision);
+    }, [props.action.message, props.action.actionName]);
```

Comment:
✋ Coming from https://github.com/Expensify/App/issues/20810

We're interested in the value of `moderationDecisions[0].decision`. We should change the dependency of the effect.

Reply:
ah nice!

Reply:
makes sense! we are also eventually switching this to only ever have one moderation decision, so we won't need this, but the fix in #20810 is great for now